engine/grid.py

```diff
@@ -300,73 +307,164 @@ def find_border_nodes(self):
+
+            Arguments:
+                start_pos: float tuple representing the starting point of the circle being plotted
+                end_pos: float tuple representing the ending point of the circle being plotted
+                center: float tuple representing the center point of the circle being plotted
+                orientation: orientation that the nodes are being plotted from, starting with the start_pos and ending
+                    at the end_pos
+            """
+            r = math.hypot(float(start_pos[0]) - center[0], float(start_pos[1]) - center[1])
+            theta_init = math.atan2(start_pos[1]-center[1], start_pos[0]-center[0])
+            theta_end = math.atan2(end_pos[1]-center[1], end_pos[0]-center[0])
+            circle_plt = []
+            if theta_end < theta_init:
+                theta_end = theta_end + 2*math.pi
+            if orientation == Orientation.CCW:
+                theta = theta_init
+                while theta < theta_end:
+                    circle_plt.append((r*math.cos(theta)+center[0], r*math.sin(theta)+center[1]))
+                    theta = theta + math.pi/12
+            elif orientation == Orientation.CW:
+                theta = theta_end
+                while theta > theta_init:
+                    circle_plt.append((r*math.cos(theta)+center[0], r*math.sin(theta)+center[1]))
+                    theta = theta - math.pi/12
+            else:
+                raise Exception("invalid orientation")
+            return circle_plt
+
         phase = WaypointPhase.DOWN
         curr_pos = self.leftmost_node_pos
+        direction = Direction.RIGHT
         waypoints = []
         waypoints.append(curr_pos)
         while (phase != WaypointPhase.TERMINATE):
-            if (phase == WaypointPhase.DOWN):
-                new_pos = (curr_pos[0]+1, curr_pos[1])
-                if curr_pos[0]+1 == rows and self.nodes[new_pos].is_active:
+            if direction == Direction.LEFT:
+                new_pos = (curr_pos[0]-1,curr_pos[1])
+                if self.nodes[new_pos].is_active:
                     waypoints.append(new_pos)
-                    right_pos = self.bottom_rightmost_node(new_pos)
-                    if right_pos is not None:
-                        waypoints.append(right_pos)
-                        curr_pos = right_pos
-                        phase = WaypointPhase.DOWN
+                    curr_pos = new_pos
+                else:
+                    left_pos = self.bottom_rightmost_node(new_pos)
+                    if left_pos is not None:
+                        new_pos = (new_pos[0]+1, new_pos[1])
+                        while new_pos[0] < left_pos[0]:
+                            while new_pos in waypoints:
+                                waypoints.remove(new_pos)
+                            new_pos = (new_pos[0]+1, new_pos[1])
+                        circle_plt = plot_circle((left_pos[0], left_pos[1]), (left_pos[0], left_pos[1]-1),
```

Comment:
based on the documentation, this doesn't seem like it'll work. assuming the traversal is top to bottom, this would (based on the documentation), make an inverted curve?

Reply:
The traversal is from bottom to top. I added the (correct?) spec to get_all_lawnmower_waypoints_adjustable

Reply:
i dont think the code i wrote is right. i'll take a look sometimes later. Current expected functionality is that it starts on the left_pos and draw a circular shape cw down towards the node directly under left_pos, but it should be starting at the bottom node and going to the top

Reply:
Can you also checks if it appends the last point at the center of the node? The last curve looks like it ended a little off the center.
<img width="892" alt="Screen Shot 2022-11-14 at 12 08 41 AM" src="https://user-images.githubusercontent.com/56476240/201580151-03d154e1-0cdf-4b39-a097-399f7b2f5bdf.png">

Reply:
done